src/test/java/core/CredentialsTest.java

```diff
@@ -1,55 +1,227 @@
+    
+    private static final String USERNAME_INPUT = "_.username";
+    private static final String PRIVATE_KEY_INPUT = "_.privateKey";
+    
+    private static final String CRED_ID = "ID";
+    private static final String CRED_USER = "user";
+    private static final String CRED_PWD = "password";
+    
     @Test @WithPlugins("ssh-credentials")
     public void createSshKeys() {
         final ManagedCredentials c = new ManagedCredentials(jenkins);
 
-        final String username = "xyz";
-        final String privKey = "1212121122121212";
-
         c.open();
         final SshPrivateKeyCredential sc = c.add(SshPrivateKeyCredential.class);
-        sc.username.set(username);
-        sc.selectEnterDirectly().privateKey.set(privKey);
+        sc.username.set(CRED_USER);
+        sc.selectEnterDirectly().privateKey.set(CRED_PWD);
+        sc.setId(CRED_ID);
         c.save();
 
         //now verify
-        jenkins.visit("credentials");
-        assertThat(find(by.input("_.username")).getAttribute("value"), is(equalTo(username)));
-        assertThat(find(by.input("_.privateKey")).getText(), is(equalTo(privKey)));
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+        verifyInputValue(CRED_ID, PRIVATE_KEY_INPUT, CRED_PWD);
     }
 
     @Test
     public void createUserPwd() {
         final ManagedCredentials c = new ManagedCredentials(jenkins);
 
-        final String username = "xyz";
-        final String password = "1212121122121212";
-
         c.open();
         final UserPwdCredential upc = c.add(UserPwdCredential.class);
-        upc.username.set(username);
-        upc.password.set(password);
+        upc.username.set(CRED_USER);
+        upc.password.set(CRED_PWD);
+        upc.setId(CRED_ID);
         c.save();
 
         //now verify
         jenkins.visit("credentials");
-        assertThat(find(by.input("_.username")).getAttribute("value"), is(equalTo(username)));
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+    }
+    
+    @Test
+    public void manageSystemScopedCredentialsTest() {
+        createUpdateDeleteTest(SYSTEM_SCOPE);
+    }
+
+    @Test
+    public void manageGlobalScopedCredentialsTest() {
+        createUpdateDeleteTest(GLOBAL_SCOPE);
+    }
+    
+    private void createUpdateDeleteTest(String systemScope) {
+        final ManagedCredentials c = new ManagedCredentials(jenkins);
+
+        // Create credential
+        c.open();
+        UserPwdCredential upc = c.add(UserPwdCredential.class);
+        upc.username.set(CRED_USER);
+        upc.password.set(CRED_PWD);
+        upc.description.set("Credential description");
+        upc.scope.select(systemScope);
+        upc.setId(CRED_ID);
+        c.save();
+
+        // verify credential was created
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+        
+        // Update credential
+        c.open();
+        upc = c.get(UserPwdCredential.class, CRED_ID);
```

Comment:
> As for the verify methods, I am not so sure. I agree that relying on the markup is not ideal, but it seems to me inevitable in many cases when implementing acceptance tests. What, I think, proves this, is that lots of tests are going to have to be fixed with Jenkins 2.0 because of changes to the markup, so I think this is just inherent to the nature of these tests. 

Even if it ware, there is no reason to make things worse. (@tfennelly might want to  correct me here, though most of the 2.0 effort addresses problems on higher level: unbundled plugins not installed, security on by default, whole parts of forms moved, text labels changed.)

> Also, when saving the credentials, the configuration page is exited. The idea behind the verify methods is to visit the page again and actually check that the inputs contains the values expected,

Agreed, the `PageArea` will do the heavy lifting for you so you do not have to locate the input manually.

> and ideally we want to check a concrete input, in the same div as the credential id.

Divs and inputs are implementation details the test should known nothing about on the first place. You want to make sure there is a *credential* scoped to some *domain* for certain *user* with given *configuration*, you do not care in how many `div`s is any of that nested.

> As for this, I could remove the exception check and just check for the id in the whole page. But I could get false positives here as well. If I create a credential inside of a domain with username bob and another credential outside of the domain with username bob as well, and then I checked for the username in the whole page, I would be checking for the first occurrence of usernmae==bob and that could not be the global one (not the one inside the domain), so I would not be checking what I want. It's up to the test implementation to reproduce this, but I just wanted to be sure that this would never happen. This actually applies to all verifies that are being performed, not just the one for the method stated.

A problem that will be fixed by referring to the `PageArea` as there will be no ambiguity.